### PR TITLE
JsonArray uniqueItems support

### PR DESCRIPTION
**Pull request by nor-ek**
I was forced to annotate `JsonNode` with `@Serializable`. I'm not sure if I should write custom serializer please check this @Kantis. 

I also wonder about creating custom violation mathod that would support all matcherss. Maybe in another task. 

By the way this PR covers:
- documentation
- tests
- code implementation

### CODE REVIEW

**Comment by nor-ek** [comment]
@Kantis this PR is ready too.

**Comment by Kantis** [comment]
Please add a test with `parseSchema` as well :)

**Comment by nor-ek** [comment]
Ahh. But there is an problem since I've added `matcher` to **JsonArray** so I need to test `parseSchema` method for `matcher` instead of `uniqueItems` which is in **JsonSchema.Builder.array.** Is that right? However have you any idea how to pass matcher in json?

Maybe I should write serializer?

**Comment by Kantis** [comment]
Actual JSON scheme should be parsed, so I think you will need to write a (de)serializer for JsonArray to be used instead of the derived serializer. I believe a few other JsonElements have explicit serializers

**Comment by nor-ek** [comment]
@Kantis done

File: kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/nodes.kt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package io.kotest.assertions.json
 
+import kotlinx.serialization.Serializable
+
+@Serializable
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/nodes.kt:5]: I believe this shouldn't be required? You can try removing it in next PR perhaps :)

**Review by Kantis — APPROVED** [review]

File: kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/nodes.kt (hunk 1 above)
Comment (no-u42) [reply kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/nodes.kt:5]: Nice catch. I thought I've deleted it

**Pull request merged**